src/main/java/hometaskStream/Book.java

```diff
@@ -0,0 +1,134 @@
+package hometaskStream;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class Book {
+    private String name;
+    private int price;
+
+    public Book() {
+        this.price = getRandomBookPrice();
+        this.name = getRandomBookName();
+    }
+
+    public Book(String name, int price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Book book = (Book) o;
+        return price == book.price && name.equals(book.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, price);
+    }
+
+    @Override
+    public String toString() {
+        return "Book name is " + this.name.toUpperCase() + " Book price is " + this.price + "$";
+    }
+
+    public static String getRandomBookName() {
+        int leftLimit = 97;
+        int rightLimit = 122;
+        int targetStringLength = 10;
+        Random random = new Random();
+        StringBuilder buffer = new StringBuilder(targetStringLength);
+        for (int i = 0; i < targetStringLength; i++) {
+            int randomLimitedInt = leftLimit + (int)
+                    (random.nextFloat() * (rightLimit - leftLimit + 1));
+            buffer.append((char) randomLimitedInt);
+        }
+        String generatedString = buffer.toString();
```

Comment:
this variable is redundant, please return the result of buffer.toString()

Reply:
Thanks, done